### ISSUE TITLE
quotemark set to single weird behaviour?

### ISSUE THREAD

**Issue by GuillaumeUnice** [palantir/tslint#4237]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.7.2
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
let k = 'bla blah\'s blah';
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "quotemark": [
      true,
      "single"
    ]
  }
}
```

#### Actual behavior
`
let k = "bla blah's blah";
`
#### Expected behavior
`
let k = 'bla blah\'s blah';
`

Just want keep the same use single quote and antislashe to protect embedded

**Comment by JoshuaKGoldberg** [comment]
@GuillaumeUnice `quotemark` will, by default, do its best to avoid adding backslashes to keep the code visually clean. Working around scattered `\`s is prone to accidentally escaping the wrong character. Would you prefer `quotemark` to have an option to be all the same slashes, regardless of `\`s?

**Comment by GuillaumeUnice** [comment]
Hi @JoshuaKGoldberg,
Yep I'd love to have an option for that, because if people want to use `\`, they could by adding an additional antislashes.
I don't really see where could there be a problem of accidentally escaping

Thanks for all

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
